store/src/main/java/org/apache/rocketmq/store/MappedFileQueue.java

```diff
@@ -461,26 +461,38 @@ public boolean commit(final int commitLeastPages) {
      */
     public MappedFile findMappedFileByOffset(final long offset, final boolean returnFirstOnNotFound) {
         try {
-            MappedFile mappedFile = this.getFirstMappedFile();
-            if (mappedFile != null) {
-                int index = (int) ((offset / this.mappedFileSize) - (mappedFile.getFileFromOffset() / this.mappedFileSize));
+            MappedFile firstMappedFile = this.getFirstMappedFile();
```

Comment:
If the first mappedFile is deleted between line 464 and line 467, this method still returns null when the requested offset is in the last mappedFile( mappedFiles.size() is less than index).
@zhouxinyu @dongeforever

Reply:
Thanks, I missed this case, please help review again -:)